ecommerce/discounts_test.py

```diff
@@ -69,3 +110,94 @@ def test_discount_factory_adjustment(discounts, products):
                 discounted_price >= 0
                 and discounted_price == discount_logic.get_product_price(product)
             )
+
+
+def perform_discount_redemption(user, discount):
+    """Redeems a discount."""
+    order = Order(purchaser=user, total_price_paid=10)
+    order.save()
+
+    redemption = DiscountRedemption(
+        redeemed_discount=discount,
+        redemption_date=now_in_utc(),
+        redeemed_order=order,
+        redeemed_by=user,
+    )
+    redemption.save()
+
+
+def test_one_time_discount(user, onetime_discount):
```

Comment:
These tests for model methods should probably live in a `models_test.py` file instead.